classes/argResolver.js

```diff
@@ -1,101 +1,76 @@
-const url = require("url");
+const Resolver = require("./Resolver");
 
 /* eslint-disable no-throw-literal, class-methods-use-this */
-module.exports = class ArgResolver {
-
-  constructor(client) {
-    Object.defineProperty(this, "client", { value: client });
-  }
+module.exports = class ArgResolver extends Resolver {
 
   async message(arg, currentUsage, possible, repeat, msg) {
     return this.msg(arg, currentUsage, possible, repeat, msg);
   }
 
   async msg(arg, currentUsage, possible, repeat, msg) {
-    if (/^\d+$/.test(arg)) {
-      const mes = await msg.channel.fetchMessage(arg).catch(() => {
-        if (currentUsage.type === "optional" && !repeat) {
-          return null;
-        }
-        throw `${currentUsage.possibles[possible].name} must be a valid message id.`;
-      });
-      return mes;
-    } else if (currentUsage.type === "optional" && !repeat) {
-      return null;
-    }
+    const message = await super.msg(arg, msg.channel);
+    if (message) return message;
+    if (currentUsage.type === "optional" && !repeat) return null;
     throw `${currentUsage.possibles[possible].name} must be a valid message id.`;
   }
 
-  async user(arg, currentUsage, possible, repeat) {
-    if ((/^<@!?\d+>$/.test(arg) && this.client.users.has(/\d+/.exec(arg)[0])) || (/\d+/.test(arg) && this.client.users.has(/\d+/.exec(arg)[0]) && arg.length > 5)) {
-      return this.client.users.get(/\d+/.exec(arg)[0]);
-    } else if (currentUsage.type === "optional" && !repeat) {
-      return null;
-    }
-    throw `${currentUsage.possibles[possible].name} must be a mention or valid user id.`;
-  }
-
   async mention(arg, currentUsage, possible, repeat) {
     return this.user(arg, currentUsage, possible, repeat);
   }
 
+  async user(arg, currentUsage, possible, repeat) {
+    const user = await super.user(arg);
+    if (user) return user;
+    if (!currentUsage) return null;
```

Comment:
It isn't... The parent class calls user internally, but since it is extended, the parent call calls the child class...